pkg/driver/helvarnet/health_test.go

```diff
@@ -0,0 +1,541 @@
+	raisedFaults := make(map[int64]bool)
+	ctx := context.Background()
+	updateDeviceFaults(ctx, status, fc, raisedFaults)
+
+	client := gen.NewDevicesApiClient(wrap.ServerToClient(gen.DevicesApi_ServiceDesc, server))
+	deviceList, err := client.ListDevices(context.TODO(), &gen.ListDevicesRequest{})
+	require.NoError(t, err)
+
+	for _, d := range deviceList.Devices {
+		checks := d.GetHealthChecks()
+		require.Equal(t, 1, len(checks))
+		require.Equal(t, 1, len(checks[0].GetFaults().CurrentFaults))
+		require.Equal(t, strconv.Itoa(BadResponseCode), checks[0].GetFaults().CurrentFaults[0].Code.Code)
+	}
+}
+
+func TestSingleHelvarnetFault(t *testing.T) {
+
+	devs := devicespb.NewCollection()
+	server := devices.NewServer(devicesServerModel{Collection: devs})
+	deviceName := "helvarnet-device-1"
+	reg := setup(devs)
+	exampleChecks := reg.ForOwner("example")
+
+	_, _ = devs.Update(&gen.Device{Name: deviceName}, resource.WithCreateIfAbsent())
+
+	check := getDeviceHealthCheck()
+	fc, err := exampleChecks.NewFaultCheck(deviceName, check)
+	require.NoError(t, err)
+	defer fc.Dispose()
+
+	status := int64(0x00000001)
+	raisedFaults := make(map[int64]bool)
+	ctx := context.Background()
+	updateDeviceFaults(ctx, status, fc, raisedFaults)
+
+	client := gen.NewDevicesApiClient(wrap.ServerToClient(gen.DevicesApi_ServiceDesc, server))
+	deviceList, err := client.ListDevices(context.TODO(), &gen.ListDevicesRequest{})
+	require.NoError(t, err)
+
+	for _, d := range deviceList.Devices {
+		checks := d.GetHealthChecks()
+		require.Equal(t, 1, len(checks))
+		require.Equal(t, 1, len(checks[0].GetFaults().CurrentFaults))
+		require.Equal(t, strconv.Itoa(0x00000001), checks[0].GetFaults().CurrentFaults[0].Code.Code)
+	}
+}
+
+func TestDoubleHelvarnetFault(t *testing.T) {
+
+	devs := devicespb.NewCollection()
+	server := devices.NewServer(devicesServerModel{Collection: devs})
+	deviceName := "helvarnet-device-1"
+	reg := setup(devs)
+	exampleChecks := reg.ForOwner("example")
+
+	_, _ = devs.Update(&gen.Device{Name: deviceName}, resource.WithCreateIfAbsent())
+
+	check := getDeviceHealthCheck()
+	fc, err := exampleChecks.NewFaultCheck(deviceName, check)
+	require.NoError(t, err)
+	defer fc.Dispose()
+
+	status := int64(0x00000011)
+	raisedFaults := make(map[int64]bool)
+	ctx := context.Background()
+	updateDeviceFaults(ctx, status, fc, raisedFaults)
+
+	client := gen.NewDevicesApiClient(wrap.ServerToClient(gen.DevicesApi_ServiceDesc, server))
+	deviceList, err := client.ListDevices(context.TODO(), &gen.ListDevicesRequest{})
+	require.NoError(t, err)
+
+	for _, d := range deviceList.Devices {
+		checks := d.GetHealthChecks()
+		require.Equal(t, 1, len(checks))
+		require.Equal(t, 2, len(checks[0].GetFaults().CurrentFaults))
+		require.Equal(t, strconv.Itoa(0x00000001), checks[0].GetFaults().CurrentFaults[0].Code.Code)
+		require.Equal(t, strconv.Itoa(0x00000010), checks[0].GetFaults().CurrentFaults[1].Code.Code)
+	}
+}
+
+func TestAddFaultThenClear(t *testing.T) {
```

Comment:
AI's are great, but they do tend to repeat themselves. Can you have a look at condensing these tests down a little to reduce the duplicate assertions and intermediate "edge" cases.